src/components/bookDetails/BookDetails.tsx

```diff
@@ -66,35 +91,103 @@ const BookDetails = () => {
     setIsOpenEditModal(false)
   }
 
+  const handleRentBook = () => {
+    RentalServices.rentBook(bookDetails.Id)
+      .then(() => {
+        toast.success('Success rent book', {
+          position: toast.POSITION.TOP_RIGHT
+        })
+        getBook()
+        getHistoryBooks()
+      })
+  }
+
+  const handleReturnBook = (id: number) => {
+    RentalServices.returnBook(id).then(() => {
+      toast.success(`${bookDetails.Title} has been returned`)
+      getHistoryBooks()
+      getBook()
+    })
+  }
+
 
   return (
-    <div className='container_for_details'>
-      <div className='container_for_img'>
-        <img className='card_img_for_book_details' src={bookDetails.Cover ? `data:image/png;base64, ${bookDetails.Cover}` : placeholderImg}/>
-      </div>
-      <div className='container_for_text'>
-        <p>Title:</p>
-        <p>{bookDetails.Title}</p>
-        <p>Description:</p>
-        <p>{bookDetails.Description}</p>
-        <p>Isbn:</p>
-        <p>{bookDetails.ISBN}</p>
-        <p>Publish Date:</p>
-        {bookDetails.PublishDate ? <p>{convertDateToString(bookDetails.PublishDate)} </p> : '' }
-        <p>Authors:</p>
-        {bookDetails.Authors &&
-            bookDetails.Authors.map((author) => ( <p key={author.Id}>{author.Firstname} {author.Lastname} </p>))
+    <>
+      <div className='container_for_details'>
+        <div className='container_for_img'>
+          <img className='card_img_for_book_details' src={bookDetails.Cover ? `data:image/png;base64, ${bookDetails.Cover}` : placeholderImg}/>
+          <div className={bookDetails.Available > 0 ? 'container_for_available_green' : 'container_for_available_red'}>
+            <p>Available:</p>
+            {bookDetails.Available}
+          </div>
+        </div>
+        <div className='container_for_text'>
+          <div className='title_container'>
+            <h4>Title:</h4>
+            <p>{bookDetails.Title}</p>
+          </div>
+          <div className='isbn_container'>
+            <h4>Isbn:</h4>
+            <p>{bookDetails.ISBN}</p>
+          </div>
+          <div className='date_container'>
+            <h4>Publish Date:</h4>
+            {bookDetails.PublishDate ? <p>{convertDateToString(bookDetails.PublishDate)} </p> : '' }
+          </div>
+          <div className='description_container'>
+            <h4>Description:</h4>
+            <p>{bookDetails.Description}</p>
+          </div>
+          <h4>Authors:</h4>
+          <div className='container_author'>
+            {bookDetails.Authors &&
+            bookDetails.Authors.map((author) => ( <p key={author.Id} className='p_tag_for_author'>{author.Firstname} {author.Lastname} </p>))
+            }
+          </div>
+        </div>
+        { !isUser
```

Comment:
i would prefer is admin or librarian over not user

Reply:
Ok i change logic